lib/charms/mysql/v0/mysql.py

```diff
@@ -979,6 +979,36 @@ def grant_privileges_to_user(
             logger.warning(f"Failed to grant privileges to user {username}@{hostname}", exc_info=e)
             raise MySQLGrantPrivilegesToUserError(e.message)
 
+    def update_user_password(self, username: str, new_password: str) -> None:
+        """Updates user password in MySQL database.
+
+        Args:
+            username: The username of user to update the password for
+            new_password: The new password to be set for the user mentioned in username arg
+
+        Raises:
+            MySQLCheckUserExistenceError if there is an issue updating the user's password
+        """
+        logger.debug(f"Updating password for {username}.")
+
+        update_user_password_commands = (
+            f"shell.connect('{self.server_config_user}:{self.server_config_password}@{self.instance_address}')",
+            f"session.run_sql(\"ALTER USER '{username}'@'%' IDENTIFIED BY '{new_password}';\")",
+            'session.run_sql("FLUSH PRIVILEGES;")',
+        )
+
+        try:
+            self._run_mysqlsh_script("\n".join(update_user_password_commands))
+        except MySQLClientError as e:
+            logger.exception(
+                f"Failed to update user password for user {username}",
+                exc_info=e,
+            )
+            raise MySQLCheckUserExistenceError(e.message)
+
+        if username == self.server_config_user:
```

Comment:
why is this needed? this MySQL class is instantiated anew every time a handler is invoked (since the charm class is also instantiated anew every time a handler is invoked)

Reply:
ack, I was not aware. was being defensive.